### ISSUE TITLE
Reduce JSON file size / Split JSON files

### ISSUE THREAD

**Issue by enrique-lozano** [f1db/f1db#9]
Currently, the resulting json after the build is over 60mb. Parsing this file in languages like python or javascript would take a long time, even opening it in the editor takes a long time. To correct this, some measures could be taken:

- Divide the JSON into several files (drivers.json, races.json, builders.json ...). I see this the simplest solution.
- Reduce the size of the IDs (numeric IDs, with acronyms...). At the IDs are called in many other places
- Eliminate duplicate fields or fields that can be calculated by other tables (for example, in the entrants we already have information about the team to which each driver belongs in each race, it would not be necessary to put that information in roots)

**Comment by marceloverdijk** [comment]
Yes, besides offering the "full" json I could also offer separate (smaller) files.

**Comment by marceloverdijk** [comment]
Hi @enrique-lozano ,

First regarding:

> Eliminate duplicate fields or fields that can be calculated by other tables (for example, in the entrants we already have information about the team to which each driver belongs in each race, it would not be necessary to put that information in the races)

Some drivers have driven for multiple entrants, and sometimes the same entrant used different engine per race. So only using the entrant reference in the race data is not sufficient.


I also did some test with splitting the json files, see here the results:

```
704B f1db-continents.json
30K  f1db-countries.json
627K f1db-drivers.json (including family relationships)
89K  f1db-constructors.json (including previous next constructors)
27K  f1db-engine-manufacturers.json
2.7K f1db-tyre-manufacturers.json
44K  f1db-entrants.json
17K  f1db-circuits.json
7.3K f1db-grands-prix.json
930K f1db-seasons.json
59M  f1db-races.json (including race results, qualifying, free practice etc.)
```

and the single json file containing all data:

```
61M  f1db-single.json
```

as expected the `f1db-races.json` contains `>95%` of all data.

So I wonder if splitting up like this is sufficient enough? What do you think?

**Comment by marceloverdijk** [comment]
Of course we could split up `races.json` further in `race-results.json`, `free-practice-1-results.json`, `free-practice-2-results.json` etc. and `seasons.json` in `season-entrants.json`, `season-driver-standings.json`, `season-constructor-standings.json`.

When generating csv it should be split up to a flattened level as well so that would be consistent.

**Comment by marceloverdijk** [comment]
Hi @enrique-lozano `v2023.0.0.alpha2` is now available with the splitted json files.
Your feedback is appreciated.

**Comment by enrique-lozano** [comment]
> Of course we could split up `races.json` further in `race-results.json`, `free-practice-1-results.json`, `free-practice-2-results.json` etc. and `seasons.json` in `season-entrants.json`, `season-driver-standings.json`, `season-constructor-standings.json`.
> 
> When generating csv it should be split up to a flattened level as well so that would be consistent.
> 
> But in that case I will need to create separate json schema's as well...

Hi! Sorry, I have been very busy these last days. I see well the division, the most practical thing would be perhaps split again the races.json (as you comment here), especially if someone would like to move from Json to CSV. Anyway it doesn't seem to me something extremely urgent. Thank you very much!

**Comment by enrique-lozano** [comment]
I just saw the new release.

This is exactly what I was looking for! What a great job, thank you very much!

**Comment by marceloverdijk** [comment]
@enrique-lozano are you using the json-schema as well?
I'm asking as I now created an additional "single "splitted" json schema, and maybe it would make more sense to have a separate json schema file per exported file (e.g. `f1db-drivers.json` + `f1db-drivers-json-schema.json`)

**Comment by enrique-lozano** [comment]
> @enrique-lozano are you using the json-schema as well? I'm asking as I now created an additional "single "splitted" json schema, and maybe it would make more sense to have a separate json schema file per exported file (e.g. `f1db-drivers.json` + `f1db-drivers-json-schema.json`)

I'm not using the json-schema but definitely yes, it makes more sense to me to separate it too. Maybe this will be useful to some people in the future.

**Comment by marceloverdijk** [comment]
For alpha3 I might split it up ;-)
